molo/surveys/static/js/survey-admin.js

```diff
@@ -52,4 +51,205 @@ $(function(){
           enableDisplayDirect();
         }
     });
+
+  window.rules = [
+    'Time Rule',
+    'Day Rule',
+    'Referral Rule',
+    'Visit Count Rule',
+    'Article Tag Rule',
+    'Query Rule',
+    'Device Rule',
+    'User Is Logged In Rule',
+    'Comment Data Rule',
+    'Profile Data Rule',
+    'Survey Submission Data Rule',
+    'Group Membership Rule',
+  ]
+
+  window.ruleIndex = {};
+  window.ruleFields = [];
+
+  window.initRuleIndex = function(){
+    rules.map(function(ruleTitle) {
+      var ruleInfo = {
+        ruleName: ruleTitle,
+        form: $('[id*="' + ruleTitle.replace(/ /g, "").toLowerCase() + '"]').filter('ul'),
+        blockIds: [],
+        updateBlock: function(){
+          this.blockIds = $.map(this.form.children('li').not('.deleted'), function(arg, index){
+            return arg.id
+          });
+        },
+        deleteBlock: function(id){
+          var index = this.blockIds.indexOf(id);
+          this.updateBlock();
+          var deletedOptionValue = this.ruleName.replace(/ /g, "") + "_" + String(index);
+          deleteRuleFromField(deletedOptionValue);
+        },
+      }
+      ruleInfo.updateBlock();
+      window.ruleIndex[ruleInfo.form.attr("id")] = ruleInfo;
+    })
+  }
+
+
+
+  window.extractRuleInfo = function(){
+    var ruleValueTextPairs = []
+    $.each(ruleIndex, function(index_outer, form){
+      $.each(form.blockIds, function(index_inner, value) {
+        ruleValueTextPairs.push({
+          "value": form.ruleName.replace(" ", "") + "_" + String(index_inner),
+          "text": form.ruleName + " " + String(index_inner + 1),
```

Comment:
Can we make this more readable for the user by appending more info?
Instead of "Time Rule 1", can we show "Time Rule 1 (03:00 - 04:00)" -- I'm just thinking that it would be more friendlier than simply having the numbers.

Reply:
The problem here is that in order to extract this data, one has to create a special function for each rule, and then to format that data. It can be done. It will just take time.

Reply:
Happy to backlog this as an improvement. Please create an issue for it.